aether-kernel/aether/kernel/api/models.py

```diff
@@ -268,6 +264,44 @@ class Meta:
         ]
 
 
+class Mapping(ExportModelOperationsMixin('kernel_mapping'), TimeStampedModel):
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4)
+    revision = models.TextField(default='1')
+    name = models.CharField(max_length=50, null=False, unique=True)
+    definition = JSONField(blank=False, null=False)
+    mappingset = models.ForeignKey(to=MappingSet, on_delete=models.CASCADE)
+    projectschemas = models.ManyToManyField(to=ProjectSchema)
+    is_active = models.BooleanField(default=True)
+    is_read_only = models.BooleanField(default=False)
+
+    # redundant but speed up queries
+    project = models.ForeignKey(to=Project, on_delete=models.CASCADE, blank=True, null=True)
+
+    def save(self, **kwargs):
+        self.project = self.mappingset.project
+        entities = self.definition.get('entities', {})
+        for entity_pk in entities.values():
```

Comment:
Reading this again I think we need to clear the field first `self.projectschemas.clear()` :grimacing: